packages/commercetools/api-client/src/api/customerCreatePasswordResetToken/index.ts

```diff
@@ -0,0 +1,35 @@
+import defaultQuery from './defaultMutation';
+import { CreatePasswordResetTokenResponse } from 'src/types/Api';
+import { Context, CustomQuery, Logger } from '@vue-storefront/core';
+import gql from 'graphql-tag';
+
+const customerCreatePasswordResetToken = async (context: Context, email: string, customQuery?: CustomQuery): Promise<CreatePasswordResetTokenResponse> => {
+  const { locale, acceptLanguage } = context.config;
+  const defaultVariables = email
+    ? {
+      locale,
+      acceptLanguage,
+      email
+    }
+    : { acceptLanguage };
+
+  const { customerCreatePasswordResetToken } = context.extendQuery(
+    customQuery, { customerCreatePasswordResetToken: { query: defaultQuery, variables: defaultVariables } }
+  );
+
+  try {
+    const result = await context.client.mutate({
+      mutation: gql`${customerCreatePasswordResetToken.query}`,
+      variables: customerCreatePasswordResetToken.variables,
+      fetchPolicy: 'no-cache'
+    }) as CreatePasswordResetTokenResponse;
+
+    return result;
+  } catch (error) {
+    Logger.error(`Cannot create password reset token. Error: ${error}`);
```

Comment:
Mostly we do not have try...catch in the sapi-client. Why is that there? Just to print more readable error than standard one?

Reply:
Yes, thanks to that it will be easier to find a bug in the future as the log shows where the error is located. Also this approach supports new way of using customQueries (introduced recently)